packages/react-instantsearch-core/src/core/context.ts

```diff
@@ -27,3 +27,14 @@ export const {
   widgetsManager: {},
   mainTargetedIndex: '',
 });
+
+export type IndexContext =
+  | {
+      targetedIndex: string;
+    }
+  | undefined;
+
+export const {
+  Consumer: IndexConsumer,
+  Provider: IndexProvider,
+} = createContext<IndexContext>(undefined);
```

Comment:
Is it really necessary? `createContext<IndexContext>()` should be equivalent?

Reply:
the TS definitions require a value there `Expected 1-2 arguments, but got 0. ts(2554)` if I remove the `undefined`